### PR TITLE
test: Recycle pip-prefix after test

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/service_annotations.go [hunk 1]
```diff
@@ -436,7 +436,10 @@ var _ = Describe("Service with annotation", func() {
 				Expect(err).NotTo(HaveOccurred())
 			}
 
-			// TODO: clean up PIPPrefix
```
Comment (lzhecheng) [inline tests/e2e/network/service_annotations.go:439]: Could you please split this `defer func()`? Move `utils.DeletePIPPrefixWithRetry` right after `utils.WaitCreatePIPPrefix`. Move `DeleteServcie` after Service creation L451.

Comment (lzhecheng) [reply tests/e2e/network/service_annotations.go:439]: After L487 Service update, please remeber to clean up the first PIP. If not, Service deletion won't consider the first PIP.

Comment (lzhecheng) [reply tests/e2e/network/service_annotations.go:439]: Could you please use `wait.Poll` for L462, L496 loops?

Comment (lodrem) [reply tests/e2e/network/service_annotations.go:439]: fixed. PTAL

File: tests/e2e/utils/network_utils.go [hunk 2]
```diff
@@ -257,6 +258,57 @@ func WaitGetPIPPrefix(
...
+	var pip network.PublicIPAddress
+
+	err := wait.Poll(10*time.Second, 5*time.Minute, func() (bool, error) {
+		prefix, err := WaitGetPIPPrefix(cli, prefixName)
+		if err != nil || prefix.PublicIPAddresses == nil || len(*prefix.PublicIPAddresses) != 1 {
+			numOfIPs := 0
+			if prefix.PublicIPAddresses != nil {
+				numOfIPs = len(*prefix.PublicIPAddresses)
+			}
+			Logf("prefix[%s] not ready with error [%s] and ip [%s]", prefixName, err, numOfIPs)
```
Comment (lzhecheng) [inline tests/e2e/utils/network_utils.go:278]:
```suggestion
			Logf("prefix [%s] not ready with error [%v] and num of IP [%d]", prefixName, err, numOfIPs)
```

File: tests/e2e/utils/network_utils.go [hunk 3]
```diff
@@ -257,6 +258,57 @@ func WaitGetPIPPrefix(
...
+			if !untilPIPCreated {
+				return true, fmt.Errorf("get pip by prefix, err: [%s], num-ip: [%d]", err, numOfIPs)
```
Comment (lzhecheng) [inline tests/e2e/utils/network_utils.go:280]:
```suggestion
				return true, fmt.Errorf("get pip by prefix, err: [%v], num-ip: [%d]", err, numOfIPs)
```

File: tests/e2e/network/service_annotations.go [hunk 4]
```diff
@@ -449,28 +448,19 @@ var _ = Describe("Service with annotation", func() {
...
-			var pip network.PublicIPAddress
-			// wait until ip created by prefix
-			for i := 0; i < 30; i++ {
-				time.Sleep(10 * time.Second)
-				prefix, err := utils.WaitGetPIPPrefix(tc, prefix1Name)
-				if err != nil || prefix.PublicIPAddresses == nil || len(*prefix.PublicIPAddresses) != 1 {
-					continue
-				}
+			pip, err := utils.WaitGetPIPByPrefix(tc, prefix2Name, true)
+			Expect(err).NotTo(BeNil())
```
[the remark this answers is not in the document]
Comment (lodrem) [reply tests/e2e/network/service_annotations.go:462]: no, we should not delete PIP here. The PIP is created by CCM.

File: tests/e2e/utils/network_utils.go (hunk 2 above)
Comment (lodrem) [reply tests/e2e/utils/network_utils.go]: fixed.

File: tests/e2e/utils/network_utils.go (hunk 3 above)
Comment (lodrem) [reply tests/e2e/utils/network_utils.go]: fixed.

**Pull request merged**